evap/results/tests/test_tools.py

```diff
@@ -190,3 +192,106 @@ def test_result_calculation_with_no_contributor_rating_question_does_not_fail(se
+                    users_seeing_contributor_own_contribution[0].add(user)
+                else:
+                    users_seeing_contributor_own_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.contributor_course_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.contributor_course_textanswer, 'full'):
+                    users_seeing_contributor_course_contribution[0].add(user)
+                else:
+                    users_seeing_contributor_course_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.contributor_all_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.contributor_all_textanswer, 'full'):
+                    users_seeing_contributor_all_contribution[0].add(user)
+                else:
+                    users_seeing_contributor_all_contribution[1].add(user)
+
+        self.assertCountEqual(general_contribution_visible_to, users_seeing_general_contribution[0])
+        self.assertCountEqual(responsible1_contribution_visible_to, users_seeing_responsible1_contribution[0])
+        self.assertCountEqual(responsible2_contribution_visible_to, users_seeing_responsible2_contribution[0])
+        self.assertCountEqual(contributor_own_contribution_visible_to, users_seeing_contributor_own_contribution[0])
+        self.assertCountEqual(contributor_course_contribution_visible_to, users_seeing_contributor_course_contribution[0])
+        self.assertCountEqual(contributor_all_contribution_visible_to, users_seeing_contributor_all_contribution[0])
+
+        self.assertEqual(len(general_contribution_delegates), len(users_seeing_general_contribution[1]))
+        self.assertEqual(len(general_contribution_delegates), 2)  # delegate1, delegate2
+        self.assertEqual(len(responsible1_contribution_delegates), len(users_seeing_responsible1_contribution[1]))
+        self.assertEqual(len(responsible1_contribution_delegates), 2)  # delegate1, contributor_course
+        self.assertEqual(len(responsible2_contribution_delegates), len(users_seeing_responsible2_contribution[1]))
+        self.assertEqual(len(responsible2_contribution_delegates), 0)
+        self.assertEqual(len(contributor_own_contribution_delegates), len(users_seeing_contributor_own_contribution[1]))
+        self.assertEqual(len(contributor_own_contribution_delegates), 2)  # delegate1, contributor_course
+        self.assertEqual(len(contributor_course_contribution_delegates), len(users_seeing_contributor_course_contribution[1]))
+        self.assertEqual(len(contributor_course_contribution_delegates), 2)  # delegate1, delegate2
+        self.assertEqual(len(contributor_all_contribution_delegates), len(users_seeing_contributor_all_contribution[1]))
+        self.assertEqual(len(contributor_all_contribution_delegates), 2)  # delegate1, contributor_course
```

Comment:
try to compare sets instead of counts. or maybe even lists because you specified the ordering in the code.

Reply:
these will stay counts when not returning the delegates but only a count in results/tools.py, right?

Reply:
yes. another thought, maybe do `self.assertTrue(len(general_contribution_delegates) == len(users_seeing_general_contribution[1]) == 2)`?